# Parameters for task sequencer
# Remember that there might be other important constants that can only be changed from the src code.

full_grasper:
  # Name of the arm
  arm_name: panda_arm
  # Name of the hand
  hand_name: right_hand
  # Arm position controller
  arm_pos_controller: position_joint_trajectory_controller
  # Arm velocity controller
  arm_vel_controller: cartesian_velocity_controller
  # Hand position controller
  hand_pos_controller: joint_trajectory_controller
  # Hand velocity controller
  hand_vel_controller: velocity_controller

  # The joint configuration for home
  home_joints: [-0.035, -0.109, -0.048, -1.888, 0.075, 1.797, -0.110] # ball
  # home_joints: [-0.109, -0.382, -0.458, -2.181, -0.160, 1.822, -0.492] # bottle
  # Transform from object frame to grasp pose frame (xyzrpy) (DEFAULT)
  # grasp_transform: [0.0, 0.0, 0.130, -3.1415, 0.0, 0.0]
  # grasp_transform: [0.028, -0.122, 0.096, 3.057, -0.272, -0.075] # ball silicone
  # grasp_transform: [0.015, -0.149, 0.110, -3.049, 0.093, -0.047]
  # grasp_transform: [0.058, -0.152, 0.152, 2.906, 0.542, -0.136] # ball giorgio
  # grasp_transform: [0.019, -0.217, 0.123, -2.460, -0.485, 0.085] # bottle
  # grasp_transform: [0.039, -0.040, 0.129, -3.123, 0.100, -0.081] # cube
  # grasp_transform: [0.039, -0.040, 0.119, -3.123, 0.100, -0.081] # apple basic idea
  grasp_transform: [0.039, -0.040, 0.109, -3.123, 0.100, -0.081] # apple basic idea
  # grasp_transform: [0.024, -0.087, 0.134, -2.501, 0.075, -0.137] # cylinder
  # Transform from grasp pose frame to pre-grasp pose frame (xyzrpy)
  pre_grasp_transform: [0.0, 0.0, -0.1, 0.0, 0.0, 0.0]
  # The joint configuration for handover
  handover_joints: [-0.101, 0.161, 0.159, -1.651, 2.023, 2.419, -0.006]
  # The threshold on tau_ext for handover hand opening
  handover_thresh: 1.0

  # The personalized grasp pose maps for different objects
  poses_map:
    "kettle" : [0.066, 0.012, 0.203, -2.973, 0.620, -0.032]
    "teddy" : [0.046, -0.238, 0.110, 3.131, -0.052, -0.400]
    "ball_thumb" : [-0.025, 0.114, 0.088, 2.879, 0.057, 0.241]
    "ball_middle" : [-0.055, 0.113, 0.075, -3.104, 0.122, -0.241]
    "ball_side" : [-0.058, 0.044, 0.027, -2.425, -0.051, 0.046]
    "two_cubes" : [-0.055, 0.113, 0.061, -3.104, 0.122, -0.241]
    "two_cubes_thumb" : [-0.025, 0.114, 0.068, 2.879, 0.057, 0.241]
    "bottle" : [0.051, -0.076, 0.077, -2.560, -0.240, 0.353]
    "helmet" : [0.120, -0.039, 0.179, 3.007, 0.200, 0.247]
    "debug" : [0.013, 0.012, 0.338, -1.512, 0.021, -0.005]
    "sponge" : [0.089, 0.008, 0.087, 3.086, 0.030, 0.057]

  # The references for different situations
  approach_ref_map:
    "sigma_d" : [0.0]
    "vel_p_d" : [0.03] # Speed for going down the local z axis
    "f_d_d" : [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]    # Forces for both approaching and pivoting (same for fully and only position constrained -> filtered by H)
  adaptive_ref_map:
    "x_d": [0.075, 0.002, 0.0, -0.003, 0.0, 0.0, 0.0]
    "f_d_d": [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]     # NOT USED!!!!
  restrain_ref_map:
    "x_d": [0.1, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    "f_d_d": [0.0, 0.0, 0.000, 0.0, 0.0, 0.0]     # Forces for restraining (same for fully and only position constrained -> filtered by H)
  lift_ref_map:
    "x_d": [0.0, 0.0, 0.0, 0.03, 0.0, 0.0, 0.0]
    "f_d_d": [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]     # Forces when lifting (NOT USED!!!!!)


# ATTENTION!!! Check if f_d_d complies with the h_matrix in adaptive_params.yaml